lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -62,36 +62,32 @@ def host_puppet_classes
 
           apipie :method, 'Checks whether a parameter value is truthly or not' do
             required :name, String, desc: 'name of the parameter'
-            optional :default_value, Object, desc: 'value to be returned if the parameter is not set on host', default: false
+            optional :default_value, Object, desc: 'value to be used if the parameter is not set on host', default: false
             returns one_of: [true, false], desc: 'Returns true if the value of a parameter can be considered as truthly, false otherwise'
             example "host_param_true?('use-ntp') #=> true"
             example "host_param_true?('use-ntp', false) #=> false if a host has no 'use-ntp' parameter"
+            example "host_param_true?('use-ntp', true) #=> true if a host has no 'use-ntp' parameter"
             see 'host_param_false?', description: '#host_param_false?', scope: Foreman::Renderer::Scope::Macros::HostTemplate
           end
           def host_param_true?(name, default_value = false)
             check_host
-            if host.params.has_key?(name)
-              Foreman::Cast.to_bool(host.params[name])
-            else
-              default_value
-            end
+            value = host.params.fetch(name, default_value)
+            Foreman::Cast.to_bool(value) == true
           end
 
           apipie :method, 'Checks whether a parameter value is falsy or not' do
             required :name, String, desc: 'name of the parameter'
-            optional :default_value, Object, desc: 'value to be returned if the parameter is not set on host', default: false
+            optional :default_value, Object, desc: 'value to be used if the parameter is not set on host', default: true
             returns one_of: [true, false], desc: 'Returns false if the value of a parameter can be considered as falsy, true otherwise'
             example "host_param_false?('use-ntp') #=> true"
-            example "host_param_false?('use-ntp', true) #=> true if a host has no 'use-ntp' parameter"
+            example "host_param_false?('use-ntp', false) #=> true if a host has no 'use-ntp' parameter"
+            example "host_param_false?('use-ntp', true) #=> false if a host has no 'use-ntp' parameter"
             see 'host_param_true?', description: '#host_param_true?', scope: Foreman::Renderer::Scope::Macros::HostTemplate
           end
-          def host_param_false?(name, default_value = false)
+          def host_param_false?(name, default_value = true)
```

Comment:
I think this still does not work properly, you changed two things, the default value and its meaning. The meaning is now the default value of the parameter if it's missing (true) which is then compared with false (the second change). Therefore this method returns false when the parameter is not found at all.

The issue you're trying to solve with this PR talks about EPEL being added by default, with this patch it is still being enabled by default.

I think that both `host_param_false?` and `host_param_true?` needs to have consistent default value for parameter that is missing. If you want to keep the change that changes the meaning of the `default_value` keep both defaulting to false

Reply:
As written in https://github.com/theforeman/foreman/pull/8295#issuecomment-778102180 I came to the conclusion that the current code "works as designed".

I've now split out the cleanup part of this into #8509 and made this PR being solely "use true? instead of !false?", thus making "enable-epel" off by default.

This is a breaking change and needs a mention in the release notes.